### ISSUE TITLE
Dirt Rally - Steam

### ISSUE THREAD

**Issue by Privation** [LibreVR/Revive#201]
I am currently having issues running the v0.8.5 version of revive with Dirt Rally on Steam.

The Steps i have followed.
1)Dump the files from the ReviveInjector.zip into the same root directory as the steam version of this game. 
2)Have Oculus home running. 
3)Have Steam Vr and Steam running. 
4)Turned off Use Desktop Game theatre. 
5) Drag Drt.exe onto ReviveInjector_x64.exe 
6) Nothing happens :-/

I have tried it with the x86 versions.  The game boots up but crashes at the start of a race.

I am running win10  AMD Radeon (TM) R9 Fury Series - with  v 16.20.1025-160602a-303160e

**Comment by KennyBarroz86** [comment]
step5: you need to drop the injector to DRT.exe :-)
i did the same mistake since it works with AC but it has to be done the other way around with DRT

**Comment by LibreVR** [comment]
Do you get the main menu in the headset?

**Comment by KennyBarroz86** [comment]
the crash you are having when starting a race can be due to resolution/graphics settings.
I had the same issue and tweaked the settings/resolution and then got it working.

both my x64 and x86 works just fine, i dont have oculus home running, just steam vr and steam.

**Comment by alerad2** [comment]
Hi there

Not getting anything in the headset just the game running on the monitor but its not crashing?

Have tried all combinations of x64, x86, steam VR beta, not beta and in game options.

win10 32GB, i7-3770K @ 4.6GHz, GTX980 368.69

**Comment by Privation** [comment]
No menu on the headset. The lenses will sometimes flicker for a second but nothing comes through.

**Comment by feathers632** [comment]
I have exactly the same problem.  Just wasted an hour loading the game over and over after trying out different settings.  There is no hint of VR with Dirt Rally (steam version).  It just displays on desktop.  All I see on the Vive is the steamvr background.  There is nothing of Dirt displayed on the vive.  I assume I have the latest librevr because I followed the link and the instructions for the steam dirt rally and the injectors. I drop the x86 files into the folder. Drop the dirt.exe onto the injector file with steamvr running.

Windows 10. Non-beta steam and non-beta steamvr.

**Comment by LibreVR** [comment]
Seems like for most people here the injection seems to fail. I'll make a Dirt Rally-specific patch as a temporary solution.

Also, please post the logs from `%APPDATA%\Revive` so I can further examine the problem with the injector.

**Comment by feathers632** [comment]
Launched injector with: ReviveInjector_x86.exe /handle 376
Injecting process handle: 376
Injecting DLL: C:\Program Files (x86)\Steam\steamapps\common\DiRT Rally\x86\openvr_api.dll
LoadLibrary found at address: 0x76904bf0
LoadLibrary failed to return module handle

**Comment by LibreVR** [comment]
Can you try dragging the executable to the `ReviveInjector_x86.exe` instead? There may be something wrong with process handle injection that is being used when the 64-bit injector delegates to the 32-bit injector.

**Comment by feathers632** [comment]
I was dragging to the x86 to start with and only tried the x64 after.  x86 injector doesn't work either.

**Comment by LibreVR** [comment]
Can you send a log after using the x86 version? Also, did you extract all files and folders? This problem can also be caused if you didn't extract the x86 folder.

**Comment by feathers632** [comment]
I've been dragging the contents out of the x86 and x64 folders and putting in the root of dirt rally folder.

So I guess that's where I went wrong?

**Comment by LibreVR** [comment]
Definitely, is there some part of the instructions that made you think that was necessary? Maybe I can reword it to make that clear?

**Comment by feathers632** [comment]
LOL.  It works now.  Yes, you will have to warn us to copy over the folders too.   It's a habit from my old days of pirate software cracks where you'd copy over the files without the folders.

Thanks for clearing that up.

**Comment by LibreVR** [comment]
I've changed the instruction to be a bit less similar to a crack, since it doesn't really matter where you extract those files as long as you extract all of them. New instructions just tell you to extract all files to a folder.

**Comment by feathers632** [comment]
So long as the folder structure in intact.  Copy as-is without taking the contents out of the two folders...

:)

**Comment by kingdom200** [comment]
Launched injector with: "C:\Program Files (x86)\Steam\steamapps\common\DiRT Rally\ReviveInjector_x86.exe" "C:\Program Files (x86)\Steam\steamapps\common\DiRT Rally\drt.exe"
Creating process: C:\Program Files (x86)\Steam\steamapps\common\DiRT Rally\drt.exe 
Injecting DLL: C:\Program Files (x86)\Steam\steamapps\common\DiRT Rally\x86\openvr_api.dll
LoadLibrary found at address: 0x768e4bf0
Injecting DLL: C:\Program Files (x86)\Steam\steamapps\common\DiRT Rally\x86\LibRevive32_1.dll
LoadLibrary found at address: 0x768e4bf0
Injected dlls succesfully

Steamvr running, htc vive showing nothing.

**Comment by LibreVR** [comment]
Log looks okay, probably still failed to inject properly. I'll have a potential workaround ready in few hours.

**Comment by feathers632** [comment]
I can tell u the thought process in my head that led to my failing to copy folders over:

"This dirt rally steam version is x86... that means I have to copy over the x86 injector and the contents of the x86 folder to dirt root".

Question:

Dirt Rally is just x86 right?  Certainly the folder names suggest that.

**Comment by LibreVR** [comment]
Correct.

**Comment by alerad2** [comment]
drt.exe to ReviveInjector_x86.exe
Launched injector with: "H:\SteamLibrary\SteamApps\common\DiRT Rally\ReviveInjector_x86.exe" "H:\SteamLibrary\SteamApps\common\DiRT Rally\drt.exe"
Creating process: H:\SteamLibrary\SteamApps\common\DiRT Rally\drt.exe 
Injecting DLL: H:\SteamLibrary\SteamApps\common\DiRT Rally\x86\openvr_api.dll
LoadLibrary found at address: 0x76c64bf0
Injecting DLL: H:\SteamLibrary\SteamApps\common\DiRT Rally\x86\LibRevive32_1.dll
LoadLibrary found at address: 0x76c64bf0
Injected dlls succesfully


drt.exe to ReviveInjector_x64.exe
Launched injector with: ReviveInjector_x86.exe /handle 356
Injecting process handle: 356
Injecting DLL: H:\SteamLibrary\SteamApps\common\DiRT Rally\x86\openvr_api.dll
LoadLibrary found at address: 0x76c64bf0
Injecting DLL: H:\SteamLibrary\SteamApps\common\DiRT Rally\x86\LibRevive32_1.dll
LoadLibrary found at address: 0x76c64bf0
Injected dlls succesfully

Runs fine on the monitor but nothing in the headset.

**Comment by kingdom200** [comment]
Dirt rally did startup yup. I was able to start a game just fine. The only problem was the vive not showing anything.

**Comment by LibreVR** [comment]
But head tracking isn't working either right? I would expect that it basically just started the game without VR support.

**Comment by alerad2** [comment]
head tracking was working on mine just no dirt in the headset.

**Comment by LibreVR** [comment]
@alerad2 Keep dirt out of your headset. ;)

I'm not sure how to solve your issue though, I wouldn't expect head tracking to work at all.

**Comment by alerad2** [comment]
I seem to be managing to keep dirt out of my headset very well :)

sorry i miss understood you, head tracking is only working in the headset showing the steamVR background, its not affecting the game on the monitor.

**Comment by Del-Dredd** [comment]
working fine here, just had Steam VR running and then dragged drt.exe with left click over to ReviveInjector_x64.exe and game started straight away and in the Vive headset as well.
Used normal button on G25 to start at start screen and it all worked fine.

Only issue seemed to be the head placement using left CTRL to reset, seemed close to windscreen, so I leaned forward, reset view and then leant back.

**Comment by LibreVR** [comment]
Can you try this patch: [ReviveDirt.zip](https://github.com/LibreVR/Revive/files/359929/ReviveDirt.zip)? Just extract it next to the executable and start Dirt Rally, no need for the injector.

Remember to make a backup of the original DLL though.

**Comment by alerad2** [comment]
no luck still the same and now just for fun my backup is corrupt :(

i will have to scan my disk and then reinstall dirt maybe a good idea anyway.

thanks for your help and if it works after the reinstall, i will report back.

**Comment by prorook** [comment]
Launched SteamVR
Downloaded and installed dirt rally
Downloaded & installed revive
Downloaded & extracted injector folders and files to my dirt directory
Unchecked "Use desktop game theater while game is active" in the dirt rally properties on steam
Dragged drt.exe to ReviveInjector_x64.exe

Launches game on desktop but I do not get anything on vive headset.

Log:
ReviveInjector_x64
Launched injector with: ReviveInjector_x86.exe /handle 368
Injecting process handle: 368
Injecting DLL: E:\Program Files (x86)\Steam\steamapps\common\DiRT Rally\x86\openvr_api.dll
LoadLibrary found at address: 0x74ae4bf0
Injecting DLL: E:\Program Files (x86)\Steam\steamapps\common\DiRT Rally\x86\LibRevive32_1.dll
LoadLibrary found at address: 0x74ae4bf0
Injected dlls succesfully

I tried extracting the files in the ReviveDirt.zip to my game directory and launching the game directly, but the same thing happened.

I don't think that I need the oculus thing installed, but I guess I'll play around with that since I saw this in the actual ReviveOverlay.log:
libpng warning: iCCP: known incorrect sRGB profile
Loaded manifest from: C:\Program Files\Revive\revive.vrmanifest
Unable to open Oculus key.

**Comment by LibreVR** [comment]
> I don't think that I need the oculus thing installed

Yes, you do actually.

Most people are probably getting this issue because they don't install Oculus Home.

**Comment by prorook** [comment]
Yeah that fixed it. Just had to install the oculus software.

**Comment by Privation** [comment]
Mines fixed.  Kept extracting the files from their respective directories instead of keeping their structure intact.

**Comment by bluedoggy** [comment]
Got MAJOR stutter on playing this with Revive. I have a pretty clean system running on a GTX1080 with windows 10. Can only launch the x86 injector and NOT the x64 injector. X64 doesn't work? Is this a possible reason it's running so poor? And yes the graphics are at the lowest quality. 
Any help would be great!

**Comment by feathers632** [comment]
What processor do you have?  Dirt rally is 32 bit only. 

There is also a shadows setting that eats into performance. Try reducing. You're not using high levels of supersampling?

**Comment by bluedoggy** [comment]
i74790. SSD drive.The system has only a few VR games on it and is pretty new. No other software as its only for VR so pretty clean.No super sample enabled. Everything is at the lowest setting. It's not a Frame rate issue. It stutters when you move your head. Reinstalled Dirt and the x64 now works but the same issue.
Maybe it's the latest NVIDIA driver?
Scratching my head as my rig is no slouch.

**Comment by bluedoggy** [comment]
Tried Assetto Corsa with 2.0 multiplier with most things on Ultra. Flawless. In fact every game I have works perfect apart from Dirt Rally?

**Comment by LibreVR** [comment]
Try the Ultra Low graphics settings in Dirt Rally, how does that work?

**Comment by bluedoggy** [comment]
Already tried it. With my spec it should breeze through with ease and some. Will try unplugging usb's reinstall drivers etc to see if there are conflicts. Last resort is a complete windows install. Strange how Assetto is working in Ultra.....

**Comment by LibreVR** [comment]
I have the same specs, but I can only run it completely smoothly at Ultra Low.

**Comment by feathers632** [comment]
You say it's not a frame rate issue but it sounds like atw not working and your fps is below 90. If your CPU is at stock speed you might try a motherboard auto overclock just to see if there's any improvement. Change nvidia driver just in case. You could install dcs 1.5 and turn on the fps indicator. That will show you the fps in vr.  Some of the missions are very heavy on CPU and you will get fps drop if there is anything wrong with your system.

**Comment by bluedoggy** [comment]
Thanks for the advice. I will try further settings. I don't understand how other guys with much lower specs with high settings are getting 90fps?
Also in the options part of dirt rally it says 59fps? Has this got anything to do with it?

**Comment by feathers632** [comment]
Have you disabled steam theater mode and overlay?  I can't test if mine shows 59fps on menu at this time.  

I think some people on steam have been running in admin mode but i don't think i needed to. 

Are you using the revive where you drop the dirt.exe onto the injector?

**Comment by LibreVR** [comment]
@bluedoggy Maybe your issue is related to this: https://steamcommunity.com/app/436320/discussions/2/366298942103623394/?ctp=6#c359543951698118816

Try creating an empty Steam account and sharing Dirt Rally with said Steam account.
